### DISCUSSION TITLE
Qual sua opinião sobre VIM/NeoVIM como "IDE"

### DISCUSSION THREAD

**Discussion by matheusbronca** [frontendbr/forum#1758] [Perguntas e Respostas]
Boa tarde pessoal, tudo bem?
Aproveitei a quarentena pra estudar um pouco de VIM, estou curtindo o processo.
Eu gostaria de saber de vocês se utilizam o VIM diariamente, e se faz sentido utilizar ele como text editor padrão. Existe alguma vantagem real em utlizar VIM puro ao invés de uma extensão com o Vscode?

**Comment by portothree** [comment]
Eu só uso vim pois deste que usei a primeira vez nunca consegui sair ou salvar o documento hahaha.

Brincadeiras  à parte. Utilizei o vim por muito tempo, até migrar para o sublime alguns anos atrás e é o editor que uso até hoje. Para não perder a memória muscular dos comandos eu utilizo a extensão Vimium nos browsers que permite navegar com os mesmos shortcuts do vim no browser.

Acredito que a vantagem se comparado ao Vscode é ser lightweight e modular, costumava ter só um arquivo .Xresources pra configurar tudo que precisava visualmente no vim.

**Comment by fdaciuk** [comment]
Pra mim, a maior vantagem de usar o vim é que você não precisa do mouse pra nada. Seus dedos ficam sempre centralizados no teclado, e com isso você consegue mais conforto, por assim dizer.

Produtividade é relativo, mas só de não ter que usar o mouse, já é um ganho pra mim :)

E usando vim + tmux, você tem tudo o que precisa no terminal. Só vai precisar do mouse quando for usar o navegador :)

**Comment by matheusbronca** [comment]
Perfeito, mas e utilizar uma solução híbrida (VSCode + Extensão VIM), não é mais vantajoso? Em termos de edição eu sei que o VIM aumenta e muito a produtividade, mas e no caso de Debugging? Eu não conheço profundamente para falar sobre o VIM, mas o VSCode fornece interfaces amigáveis pra tal atividade.

**Comment by fdaciuk** [comment]
Acho que é mais questão de gosto: o que você precisa de debugging que o VS Code faz? Quando eu quero debugar algo, posso usar o próprio navegador pra isso, dependendo do caso :)

**Comment by matheusbronca** [comment]
No caso de desenvolvimento Mobile com Flutter por exemplo, o VSCode fornece um pacote de ferramentas visuais, o Flutter Inspector + Dev Tools, neste caso em expecífico creio que não seria possível reproduzir algo semelhante no VIM. (ou seria?)

Obrigado pelos comentários

**Comment by fdaciuk** [comment]
Saquei.. nesse caso, não sei se seria muito prático usar o Vim, [mas dá](https://www.dannyadam.com/blog/2019/05/debugging-in-vim/) :)

**Comment by expalmer** [comment]
Gente sempre quis ir pro Vim, mas nunca consigo configurar direito. Por acaso vocês indicam algum tutorial para deixar configurado para quem vem de sublime/vscode ?

**Comment by matheusbronca** [comment]
> Gente sempre quis ir pro Vim, mas nunca consigo configurar direito. Por acaso vocês indicam algum tutorial para deixar configurado para quem vem de sublime/vscode ?

Tem esse tutorial do Pagar.me que eu achei super bacana:
 https://www.youtube.com/watch?v=kZDT10nFiTY

Espero que ajude, tmj :)

**Comment by onlurking** [comment]
Particularmente uso VSCode com a extensão do [VSVim](https://marketplace.visualstudio.com/items?itemName=JaredParMSFT.VsVim) no trabalho, e uso o Neovim mesmo para anotações (uso vimwiki e notational-fzf-vim) e outras edições um pouco mais simples editar commits message/rebase.

Eu particularmente gosto do Vim justamente pela edição e movimentação dele(text objects, motions) e algumas features como macros e comandos `ex`(norm, argdo), mas tenho tudo isso no VSCode, então nesse caso prefiro ser pragmático do que ser purista (e a parte de staging de commits do VS é maravilhoso).

Fora que faz pelo menos uns 3 anos que edito meu .vimrc e até hoje não estou satisfeito, e eu preciso trabalhar xP

Tem esse livro aqui que é muito bom pra aprender Vim de quem vem do VS Code (é focado no VSVim): 
- [Boost Your Coding Fu With VSCode and Vim](https://www.barbarianmeetscoding.com/boost-your-coding-fu-with-vscode-and-vim/table-of-contents)
